src/index.js

```diff
@@ -156,161 +168,139 @@ app.get('/travis.org/*', (req, res, next) => {
+    const build = await travis.getLatestBranchBuild(branch);
     const sauce = new SauceClient(sauceUser, accessKey);
-    // eslint-disable-next-line promise/no-nesting
-    const jobs = travis.getLatestBranchBuild(branch).then(build => {
-      return sauce.getTravisBuildJobs(build);
+    const sauceJobs = await sauce.getTravisBuildJobs(build);
+    const body = await getSauceBadgeBody({
+      req,
+      sauce,
+      source: 'api',
+      sauceJobs
     });
-    return handleSauceBadge(req, res, sauce, 'api', jobs);
-  });
-});
-
-// eslint-disable-next-line max-statements
-app.get('/size/:source/*', (req, res) => {
-  res.status(200);
-  res.set('Content-Type', 'image/svg+xml');
-  if (cachingEnabled) {
-    res.set('Cache-Control', cacheHeader);
-  }
-  res.flushHeaders();
 
-  console.log(`Incoming request from referrer: ${req.get('Referrer')}`);
+    sendResponse({ res, body });
+  })
+);
 
-  const source = req.params.source;
-  const path = req.params[0];
-  const color = req.query.color || 'brightgreen';
-  const options = { gzip: req.query.gzip === 'true' };
-  const query = { style: req.query.style };
-  let url;
-  // Express' path-to-regexp business is too insane to easily do this above.
-  if (path.length > 0) {
-    if (source === 'github') {
-      url = `https://raw.githubusercontent.com/${path}`;
-    } else if (source === 'npm') {
-      url = `https://unpkg.com/${path}`;
+// eslint-disable-next-line max-statements
+app.get(
+  '/size/:source/*',
+  asyncMiddleware(async (req, res) => {
+    const source = req.params.source;
+    const path = req.params[0];
+    const color = req.query.color || 'brightgreen';
+    const options = { gzip: req.query.gzip === 'true' };
+    const query = { style: req.query.style };
+    let url;
+    // Express' path-to-regexp business is too insane to easily do this above.
+    if (path.length > 0) {
+      if (source === 'github') {
+        url = `https://raw.githubusercontent.com/${path}`;
+      } else if (source === 'npm') {
+        url = `https://unpkg.com/${path}`;
+      }
     }
-  }
-  const label = req.query.label || (options.gzip ? 'size (gzip)' : 'size');
-  // eslint-disable-next-line promise/catch-or-return
-  getFileSize(url, options)
-    .then(size => {
-      return getShieldsBadge(label, size, color, query);
-    })
-    .catch(err => {
-      console.error(`Error: ${err}`);
-      return getShieldsBadge(label, 'error', 'lightgrey', query);
-    })
-    // eslint-disable-next-line promise/always-return
-    .then(body => {
-      res.write(body);
-      res.end();
-    });
-});
+    const label = req.query.label || (options.gzip ? 'size (gzip)' : 'size');
 
-app.get('/browsers', (req, res) => {
-  res.status(200);
-  res.set('Content-Type', 'image/svg+xml');
-  if (cachingEnabled) {
-    res.set('Cache-Control', cacheHeader);
-  }
-  res.flushHeaders();
+    const body = await getFileSize(url, options)
+      .then(size => getShieldsBadge(label, size, color, query))
```

Comment:
Any advantage of the promise chain here vs. `try/catch`?

Reply:
Great question! I was originally refactoring it, but catching two separate async calls in a promise chain seemed easier. Otherwise, I'd have a big try catch with two `let` variables outside that just seemed messier...

Happy to change if you think better?

Reply:
ew, yeah I see that now 😂 better as-is